cli/src/chain_spec/mod.rs

```diff
@@ -71,8 +73,21 @@ pub struct NetworkKeys {
 	pub root_key: AccountId,
 }
 
+/// Node `ChainSpec` extensions.
+///
+/// Additional parameters for some Substrate core modules,
+/// customizable from the chain spec.
+#[derive(Default, Clone, Serialize, Deserialize, ChainSpecExtension)]
+#[serde(rename_all = "camelCase")]
+pub struct Extensions {
```

Comment:
by reintroducing this you will need to update the genesis.jsons with these fields `fork_blocks` and `bad_blocks`.
Can test this by trying to run a node with `--chain=<network>.raw.json`

Reply:
I had to reintroduce these to make the tests build. The unit struct as the second template parameter for GenericChainSpec has the issue of not implementing ChainSpecExtension which is needed in some client service functions.
I will fix the genesis files and test them.

Reply:
@holygits Without any change to the genesis file, I could connect and sync with Azalea.